src/Prometheus/RenderTextFormat.php

```diff
@@ -53,11 +48,31 @@ private function renderSample(MetricFamilySamples $metric, Sample $sample): stri
      * @param string $v
      * @return string
      */
-    private function escapeLabelValue($v): string
+    private function escapeLabelValue(string $v): string
+    {
+        return str_replace(["\\", "\n", "\""], ["\\\\", "\\n", "\\\""], $v);
+    }
+
+    /**
+     * @param string[]  $labelNames
+     * @param Sample $sample
+     *
+     * @return string[]
+     */
+    private function escapeAllLabels(array $labelNames, Sample $sample): array
     {
-        $v = str_replace("\\", "\\\\", $v);
-        $v = str_replace("\n", "\\n", $v);
-        $v = str_replace("\"", "\\\"", $v);
-        return $v;
+        $escapedLabels = [];
+
+        $labels = array_combine(array_merge($labelNames, $sample->getLabelNames()), $sample->getLabelValues());
+
+        if ($labels === false) {
+            return $escapedLabels;
```

Comment:
I would prefer to return an empty array here, because it is easier to reed.
```suggestion
            return [];

Reply:
I had this originally as well, agree its more readable